txp/Pos/Txp/Toil/Utxo/Functions.hs

```diff
@@ -250,5 +266,9 @@ rollbackTxUtxo
 rollbackTxUtxo (txAux, undo) = do
     let tx@UnsafeTx {..} = taTx txAux
     let txid = hash tx
-    mapM_ utxoDel $ take (length _txOutputs) $ map (TxIn txid) [0..]
-    mapM_ (uncurry utxoPut) $ NE.zip _txInputs undo
+    mapM_ utxoDel $ take (length _txOutputs) $ map (TxInUtxo txid) [0..]
+    mapM_ (uncurry utxoPut) $ concatMap knownInputAndUndo $ toList $ NE.zip _txInputs undo
+  where
+    knownInputAndUndo (_, Nothing)         = []
+    knownInputAndUndo (TxInUnknown _ _, _) = []
+    knownInputAndUndo (inp, Just u)        = [(inp, u)]
```

Comment:
@gromakovsky do you want `error`?
tbh it's so dangerous. a user might update node and load block from db and decode txin which was unknown earlier but there isn't still undo for this txin.